source/common/stats/isolated_store_impl.h

```diff
@@ -268,9 +236,126 @@ class IsolatedStoreImpl : public StoreImpl {
   IsolatedStatsCache<TextReadout> text_readouts_;
   RefcountPtr<NullCounterImpl> null_counter_;
   RefcountPtr<NullGaugeImpl> null_gauge_;
-  ScopeSharedPtr default_scope_;
+  mutable ScopeSharedPtr lazy_default_scope_;
```

Comment:
Why is this lazy-created? Why is there ever not a root scope, and so why can't it be created upon construction? It seems like some other parts could be simpler if `rootScope()` never has to create it.

Reply:
It makes it easier to subclass IsoaltedStoreImpl, allowing the infrastructure to create and maintain scopes to be shared, using a virtual method to construct a Scope of the appropriate type.

I will add a comment.

Reply:
Perhaps, but it's a little messy since there are two levels of subclassing of this in our test stores, and so the intermediate one (which is often used directly) would I guess need to make one and then have it erased by its subclass.

I felt it was cleaner to do this lazily after all construction is done